### ISSUE TITLE
feature request: page / group toggle for checkboxes

### ISSUE THREAD

**Issue by Arteneko** [giuspen/cherrytree#1863]
When dealing with a lot of checkboxes, e.g. as a regularly used planning sheet, it would be cool to be able to mass-toggle the state of a group of checkboxes (or of the whole page).

What do you think about the feasability of this?

**Comment by giuspen** [comment]
you mean the so called TODO lists?

**Comment by Arteneko** [comment]
I mean this, if that's the todo-list (which it probably is).

![a series of checkboxes prefixing a few lines of text](https://user-images.githubusercontent.com/23583792/142086016-91d94a15-a41a-4d47-bd9d-50624868f6fc.png)

**Comment by metal450** [comment]
Same wish here - I have some notes that are i.e. a monthly ToDo list, & each time I have to go in and manually uncheck all the boxes one at a time to "reset" the list. Would be hugely helpful if there were a quick way to auto-check/auto-uncheck all of them within the current page.

**Comment by Kansai53** [comment]
Yes, would make clearing out grocery lists a lot quicker and easier 🙏 🙏

**Comment by Ricent82** [comment]
I have the same issue. Used to use OneNoteGem to do this when I was using OneNote, but while I way prefer cross-platform CherryTree, resetting my todo lists takes a lot of manual rote unchecking. Wish there was a one-button "uncheck all in note" option

**Comment by giuspen** [comment]
If the issue is only resetting to non ticked, there is a quick way to achieve that: just select all the elements you want to reset, then change the type of list to for example bulleted, then change again to Todo list

**Comment by Ricent82** [comment]
That only works if there's no non-checkbox lines or content interspersed in the list. Otherwise, it will turn every line into a checkbox, rather than just unchecking the checkboxes that are already there.

**Comment by Kansai53** [comment]
Could I ping on this again just since it's been another 12mo? Oh how I wish there were a way to not have to manually uncheck 50 checkboxes to reset my weekly todo list ;)

**Comment by gitvectors** [comment]
I follow theese threads not to request features but to learn how external scripts  can meet different user needs. 

_Provided that_  you are using *.ctd mode  (you can always convert from other modes) _then_ you can use XML parsing tools to parse your *.ctd/xml code .
This link refers to XML parsing.
https://docs.python.org/3/library/xml.etree.elementtree.html
An easier way of XML parsing is to use an XML editor such as free XMLCopyEditor.  Change your checkboxes using xquery.
A more general approach I'm considering it to develop a python plugin for Albert (in Ubuntu) or Alfred (in Mac) to conduct operations on multlple *.ctd documents. Later development task. There is a Windows alternative.

In a word .. automation scripts.

**Comment by Kansai53** [comment]
I'm using multi-file (which is xml). Seems pretty hacky/brittle though - would have to know the node ids to update (to find the appropriate xml files), script needs to ask which one to navigate to in order to do the checkboxes, it would have to know which characters are used for checkboxes (from ct settings), etc. Doesn't scale to other users. Is it technically possible? Yeah. But with all that effort...have I really saved time over just clicking them manually? Probably not...

**Comment by dbwiz** [comment]
Why don't you just search and replace (CTRL+H)   "☑ " with "☐ "?
If you don't use replace often its even the default next time you use it:   CTRL+H, then "Ok"

**Comment by metal450** [comment]
That's....a great idea :rofl: :man_facepalming:
